pip/req/req_set.py

```diff
@@ -439,6 +461,11 @@ def _prepare_file(self, finder, req_to_install):
             # # vcs update or unpack archive # #
             # ################################ #
             if req_to_install.editable:
+                if require_hashes:
+                    raise InstallationError(
+                        'The editable requirement %s cannot be installed when '
+                        'requiring hashes, because there is no single file to '
+                        'hash.' % req_to_install)
```

Comment:
I assume later commits fsck fine, correct? I'd be willing to accept that broken commits can't be used as hash-checked VCS checkouts by pip. But the point is moot because SHA1 will probably be cost-effectively breakable in a few years.